kernel/src/main/java/org/kframework/compile/AddKoreAttributes.java

```diff
@@ -0,0 +1,331 @@
+            }
+        }
+        Seq<K> setArgsSeq = JavaConverters.iterableAsScalaIterable(setArgs).toSeq();
+
+        KLabel equalsLabel = KLabel(KLabels.ML_EQUALS.name(), Sorts.Bool(), sortParam);
+        Rule ceilMapRule =
+                Rule(
+                        KRewrite(
+                                KApply(ceilMapLabel,
+                                        KApply(concatProd.klabel().get(),
+                                                KApply(elementProd.klabel().get(),
+                                                        setArgsSeq,
+                                                        Att.empty()
+                                                ),
+                                                restMapSet
+                                        )
+                                )
+                                ,
+                                KApply(andLabel,
+                                        KApply(equalsLabel,
+                                                KApply(prod.klabel().get(),
+                                                        setArgs.get(0),
+                                                        restMapSet
+                                                ),
+                                                BooleanUtils.FALSE
+                                        ),
+                                        setArgsCeil
+                                )
+                        )
+                        , BooleanUtils.TRUE
+                        , BooleanUtils.TRUE
+                        , Att.empty().add(Att.SIMPLIFICATION())
+                );
+        rules.add(ceilMapRule);
+    }
+
+    private boolean isRealHook(Att att) {
+        String hook = att.get(Att.HOOK());
+        if (hook.startsWith("ARRAY.")) {
+            return false;
+        }
+        if (options.hookNamespaces.stream().anyMatch(ns -> hook.startsWith(ns + "."))) {
+            return true;
+        }
+        return Hooks.namespaces.stream().anyMatch(ns -> hook.startsWith(ns + "."));
+    }
+
+    public static final Production INJ_PROD = Production(KLabel(KLabels.INJ, Sort("S1"), Sort("S2")), Sort("S2"), Seq(NonTerminal(Sort("S1"))), Att());
+
+    private Production production(KApply term) {
+        return production(term, false);
+    }
+    private Production production(KApply term, boolean instantiatePolySorts) {
+        KLabel klabel = term.klabel();
+        if (klabel.name().equals(KLabels.INJ))
+            return instantiatePolySorts ? INJ_PROD.substitute(term.klabel().params()) : INJ_PROD;
+        Option<scala.collection.Set<Production>> prods = module.productionsFor().get(klabel.head());
+        if (!(prods.nonEmpty() && prods.get().size() == 1))
+            throw KEMException.compilerError("Expected to find exactly one production for KLabel: " + klabel + " found: " + prods.getOrElse(Collections::Set).size());
```

Comment:
For some reason, this Exception is thrown due to finding 0 production for KLabel.

Reply:
I can't tell you why that might be without knowing what klabel it's throwing the error for. In general it's always better to share the entire exact error message rather than to just summarize in a way that leaves information out.